### PR TITLE
Convert _math_style_dict into an Enum.

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

Instead of a weird `dict` thing.

## PR Checklist

- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [n/a] New features are documented, with examples if plot related.
- [n/a] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and `pydocstyle<4` and run `flake8 --docstring-convention=all`).
- [n/a] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [n/a] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Review by story645 — APPROVED** [review]

File: lib/matplotlib/_mathtext.py [hunk 1]
```diff
@@ -1944,8 +1945,11 @@ class Parser:
     The grammar is based directly on that in TeX, though it cuts a few corners.
     """
 
-    _math_style_dict = dict(displaystyle=0, textstyle=1,
-                            scriptstyle=2, scriptscriptstyle=3)
+    class _MathStyle(enum.Enum):
+        DISPLAYSTYLE = enum.auto()
+        TEXTSTYLE = enum.auto()
+        SCRIPTSTYLE = enum.auto()
+        SCRIPTSCRIPTSTYLE = enum.auto()
```
Comment (story645) [inline lib/matplotlib/_mathtext.py:1953]: Where are these two used?

**Review by story645 — APPROVED** [review]

File: lib/matplotlib/_mathtext.py (hunk 1 above)
Comment (QuLogic) [reply lib/matplotlib/_mathtext.py:1953]: Hmm, apparently nowhere. My guess is they were defined for compatibility with TeX, but never implemented. Looking at commit messages of #8151, I think that was the author's goal but maybe never got back to it.

**Comment by jklymak** [comment]
So we are swapping a weird dict thing for a weird enum thing? :wink:  Was there something that motivated this?

**Comment by QuLogic** [comment]
Modernization, but then I didn't find any other things to convert to enum, so I figured I'd open the PR before it became unmergeable (which is already did once).

But this does mean no typos, because it can be linted, and works with typing, if we ever get around to that.

**Comment by anntzer** [comment]
I guess `(DISPLAYSTYLE, ...) = range(4)` (*in the Enum definition*, not as globals) may be less magical for @jklymak? :) (I slightly prefer it too, it's actually shorter to type...)

**Comment by jklymak** [comment]
I mean, I dont' really mind, I just don't see why this is particularly better for something that isn't broken

**Review by dopplershift — APPROVED** [review]

**Pull request merged**